hibernate-core/src/main/java/org/hibernate/collection/spi/AbstractPersistentCollection.java

```diff
@@ -286,7 +286,11 @@ else if ( !session.isConnected() ) {
 				 Whenever the collection lazy loading is triggered during the loading process,
 				 closing the connection will cause an error when RowProcessingStateStandardImpl#next() will be called.
 				 */
-				if ( !session.isTransactionInProgress() && session.getPersistenceContext().isLoadFinished() ) {
+				final PersistenceContext persistenceContext = session.getPersistenceContext();
+				if ( !session.isTransactionInProgress()
```

Comment:
I think this can be simplified, from
```java
&& ( !persistenceContext.hasLoadContext()
		|| ( persistenceContext.hasLoadContext()
			&& persistenceContext.getLoadContexts().isLoadingFinished() ) )
```
to
```java
&& ( !persistenceContext.hasLoadContext()
		|| persistenceContext.getLoadContexts().isLoadingFinished() )
```

Reply:
It surely can, but I didn't want to be pedantic. We will do this eventually during cleanups.